framework/src/media/utils/MediaUtils.cpp

```diff
@@ -511,6 +558,71 @@ bool header_parsing(unsigned char *buffer, unsigned int bufferSize, audio_type_t
 	return true;
 }
 
+#ifdef CONFIG_MPEG2_TS
+bool ts_parsing(unsigned char *buffer, unsigned int bufferSize, audio_type_t *audioType, unsigned int *channel, unsigned int *sampleRate, audio_format_type_t *pcmFormat)
+{
+	// create temporary ts parser
+	auto tsDemuxer = media::TSDemuxer::create();
+	if (!tsDemuxer) {
+		meddbg("TSDemuxer::create failed\n");
+		return false;
+	}
+
+	// push the given (ts) data into demux buffer
+	size_t ret = tsDemuxer->pushData(buffer, bufferSize);
+	if (ret < bufferSize) {
+		medwdbg("TSDemuxer accept part of data %u/%u\n", ret, bufferSize);
+	}
+
+	// pre parse to get PSI
+	if (tsDemuxer->prepare() < 0) {
+		meddbg("TSDemuxer parse failed\n");
+		return false;
+	}
+
+	// get programs in ts, usually we select the 1th one as default.
+	std::vector<unsigned short> programs;
+	tsDemuxer->getPrograms(programs);
+	medvdbg("There's %lu programs in the given transport stream\n", programs.size());
+	if (programs.empty()) {
+		meddbg("TSDemuxer didn't find any program! Failed!\n");
+		return false;
+	}
+
+	// get audio type (from PMT component stream type field)
+	*audioType = tsDemuxer->getAudioType((void *)&programs[0]);
+
+	// get ES data (we expect the given ts data is enough to form a PES packet)
+	unsigned char audioES[64]; // 64 bytes should be enough for header parsing
+	ssize_t audioESLen = tsDemuxer->pullData(audioES, sizeof(audioES), (void *)&programs[0]);
+	if (audioESLen < 0) {
+		meddbg("TSDemuxer get ES data failed!\n");
+		return false;
+	}
+
+	return header_parsing(audioES, (unsigned int)audioESLen, *audioType, channel, sampleRate, pcmFormat);
```

Comment:
It needs to initialize pcmFormat

Reply:
I initialized it before calling stream_parsing(...)